FirebaseAppCheck/Tests/Unit/Swift/AppCheckAPITests.swift

```diff
@@ -0,0 +1,200 @@
+//
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// MARK: This file is used to evaluate the experience of using Firebase APIs in Swift.
+
+import Foundation
+
+import FirebaseCore
+import FirebaseAppCheck
+
+final class AppCheckAPITests: NSObject {
+  func usage() {
+    // MARK: - AppAttestProvider
+
+    #if TARGET_OS_IOS
+      if #available(iOS 14.0, *) {
+        if let app = FirebaseApp.app(), let provider = AppAttestProvider(app: app) {
+          provider.getToken { token, error in
+            // ...
+          }
+        }
+      }
+    #endif // TARGET_OS_IOS
+
+    // MARK: - AppCheck
+
+    // `AppCheckTokenDidChange` & associated notification keys
+    _ = NotificationCenter.default
+      .addObserver(
+        forName: .AppCheckTokenDidChange,
+        object: nil,
+        queue: .main
+      ) { notification in
+        _ = notification.userInfo?[AppCheckTokenNotificationKey]
+        _ = notification.userInfo?[AppCheckAppNameNotificationKey]
+      }
+
+    // Retrieving an AppCheck instance
+    _ = AppCheck.appCheck()
+
+    if let app = FirebaseApp.app() {
+      _ = AppCheck.appCheck(app: app)
+    }
+
+    // Get token
+    AppCheck.appCheck().token(forcingRefresh: false) { token, error in
+      if let _ /* error */ = error {
+        // ...
+      } else if let _ /* token */ = token {
+        // ...
+      }
+    }
+
+    // Get token (async/await)
+    #if swift(>=5.5)
+      if #available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *) {
```

Comment:
Any idea why GitHub is marking macCatalyst as red?

Reply:
No idea, it's weird. And yep, I'm developing on Xcode 13b1.